### PR TITLE
Add example of using borsh serializer

### PR DESCRIPTION
**Pull request by mikedotexe**
This is related to some work in service to #9 
Will try to be pushing smallish PRs as I carve time.

### CODE REVIEW

File: docs/contract-interface/serialization-interface.md [hunk 1]
```diff
@@ -35,6 +35,26 @@ pub fn sum_borsh(#[serializer(borsh)] a: u32, #[serializer(borsh)] b: u32) -> u3
...
+A simple demonstration of getting a [Borsh-serialized](https://borsh.io), base64-encoded value from a unit test:
+
+```rust reference
+https://github.com/mikedotexe/rust-status-message/blob/03781079d1716584d114bec294de00f5336cd20d/src/lib.rs#L106-L114
+```
+
+The following snippet shows a simple function that takes this value from a frontend or CLI. Note: this method doesn't have a return value, so the `#[result_serializer(borsh)]` isn't needed.
+
+```rust reference
+https://github.com/mikedotexe/rust-status-message/blob/03781079d1716584d114bec294de00f5336cd20d/src/lib.rs#L41-L55
```
Comment (ChaoticTempest) [inline docs/contract-interface/serialization-interface.md:49]: Formatting for this one has one too many closing bracket at the end. Might just be my OCD, but would look nice to not have it there

Comment (mikedotexe) [reply docs/contract-interface/serialization-interface.md]: good catch, fixed!

**Comment by mikedotexe** [comment]
@ChaoticTempest I added another page to look at, too

File: docs/cross-contract/callbacks.md [hunk 2]
```diff
@@ -4,10 +4,48 @@ sidebar_position: 2
...
 
-<!-- Why do we need them and when should they be used -->
+The common pattern with cross-contract calls is to call a method on an external smart contract, use `.then` syntax to specify a callback, and then retrieve the result or status of the promise. The callback will typically live inside the same, calling smart contract. There's a special macro used for the callback function, which is [#[private]](https://docs.rs/near-sdk-core/latest/near_sdk_core/struct.AttrSigInfo.html#structfield.is_private). We'll see this pattern in the example below.
 
-TODO
+The following example demonstrates two common approaches to callbacks using the high-level cross-contract approach. When writing high-level cross-contract calls, special [traits](https://doc.rust-lang.org/rust-by-example/trait.html) are set up that serve as interfaces for the smart contract being called, and (typically) the current contract doing the calling, where callback logic will live. The second trait is, by convention, referred to as `ext_self`. Here are the two traits used in a simple example:
```
Comment (ChaoticTempest) [inline docs/cross-contract/callbacks.md:13]: nit: I was trying to unravel the mid sentence in my mind for a bit so I think its worth rewording it. I believe this is what you meant and hopefully its clearer:

```suggestion
The following example demonstrates two common approaches to callbacks using the high-level cross-contract approach. When writing high-level cross-contract calls, special [traits](https://doc.rust-lang.org/rust-by-example/trait.html) are set up as interfaces for the smart contract being called, and (typically) the current contract doing the calling (where callback logic will live). The second trait is, by convention, referred to as `ext_self`. Here are the two traits used in a simple example:
```

The previous one wasn't clear where the subject of the statement "where callback logic will live" was, IMO.

**Review by ChaoticTempest — APPROVED** [review]

File: docs/cross-contract/callbacks.md [hunk 3]
```diff
@@ -4,10 +4,48 @@ sidebar_position: 2
 
 # Callbacks
 
-<!-- What are callbacks -->
+NEAR Protocol is a sharded, proof-of-stake blockchain that behaves differently than proof-of-work blockchains. When interacting with a native Rust (compiled to Wasm) smart contract, cross-contract calls are asynchronous. Callbacks are used to either get the result of a cross-contract call or tell if a cross-contract call has succeeded or failed.
```
Comment (austinabell) [inline docs/cross-contract/callbacks.md:7]: I mean, the PoW vs PoS doesn't affect the need for asynchronous calls, does it? If anything, a sharded PoW chain would gain more value from async calls, I think.

File: docs/cross-contract/callbacks.md (hunk 2 above)
Comment (mikedotexe) [reply docs/cross-contract/callbacks.md]: Thank you!

File: docs/contract-interface/serialization-interface.md [hunk 4]
```diff
@@ -35,6 +35,26 @@ pub fn sum_borsh(#[serializer(borsh)] a: u32, #[serializer(borsh)] b: u32) -> u3
...
+A simple demonstration of getting a [Borsh-serialized](https://borsh.io), base64-encoded value from a unit test:
+
+```rust reference
+https://github.com/mikedotexe/rust-status-message/blob/03781079d1716584d114bec294de00f5336cd20d/src/lib.rs#L106-L114
+```
+
+The following snippet shows a simple function that takes this value from a frontend or CLI. Note: this method doesn't have a return value, so the `#[result_serializer(borsh)]` isn't needed.
+
+```rust reference
+https://github.com/mikedotexe/rust-status-message/blob/03781079d1716584d114bec294de00f5336cd20d/src/lib.rs#L41-L54
```
Comment (austinabell) [inline docs/contract-interface/serialization-interface.md:49]: This reference seems wrong, there are some syntactic issues but I'm very confused on why it's being parsed differently for wasm and non-wasm arch. This seems like a bad pattern, what's the reason for it?

Comment (mikedotexe) [reply docs/contract-interface/serialization-interface.md]: I've pushed a commit changing the reference here. The cfg deal was the make sure unit tests could run but I agree that adds complexity where we don't need it.

File: docs/contract-interface/serialization-interface.md [hunk 5]
```diff
@@ -35,6 +35,26 @@ pub fn sum_borsh(#[serializer(borsh)] a: u32, #[serializer(borsh)] b: u32) -> u3
...
+A simple demonstration of getting a [Borsh-serialized](https://borsh.io), base64-encoded value from a unit test:
+
+```rust reference
+https://github.com/mikedotexe/rust-status-message/blob/03781079d1716584d114bec294de00f5336cd20d/src/lib.rs#L106-L114
+```
+
+The following snippet shows a simple function that takes this value from a frontend or CLI. Note: this method doesn't have a return value, so the `#[result_serializer(borsh)]` isn't needed.
+
+```rust reference
+https://github.com/near/rainbow-bridge-rs/blob/b9096189af828d3684c9124dbd0c0072a35cd9f7/mintable-fungible-token/src/lib.rs#L399-L401
```
Comment (austinabell) [inline docs/contract-interface/serialization-interface.md:49]: Are you sure this is what you meant to link? https://github.com/near/rainbow-bridge-rs/blob/b9096189af828d3684c9124dbd0c0072a35cd9f7/mintable-fungible-token/src/lib.rs#L399-L401

Comment (mikedotexe) [reply docs/contract-interface/serialization-interface.md]: I think I made some silly approaches here, pushed a change that should be much better, thanks for the comments

**Review by austinabell — APPROVED** [review]

**Pull request merged**